core/src/error.rs

```diff
@@ -2052,6 +2061,25 @@ impl IntoDiagnostics<FileId> for TypecheckError {
                         ]),
                 ]
             }
+            TypecheckError::FlatTypeInTermPosition { flat, pos } => {
+                vec![
+                    Diagnostic::error()
+                        .with_message(
+                            "types containing user-defined contracts cannot be converted into contracts",
```

Comment:
I feel this message is a bit vague. But to be honest, this error is quite specific to our current implementation choices, so it's not too easy to get right in a succinct way. We can meditate on this and improve on later, so let's not block this PR on this.